dateutil/parser.py

```diff
@@ -398,6 +398,17 @@ def has_month(self):
     def has_day(self):
         return self.dstridx is not None
 
+    def could_be_day(self, value):
+        if self.has_day:
+            return False
+        elif not self.has_month:
+            return 1 <= value <= 31
+        elif not self.has_year:
+            # Be permissive, assume leapyear
+            return 1 <= value <= monthrange(2000, self.month)[1]
```

Comment:
Weird choice of an exemplary leap year, since this is one of the weird leap years.

Is it meaningful to add a test that will hit this branch for leap and non-leap years?

Reply:
Fair point.  I'm happy to use 1996.

> Is it meaningful to add a test that will hit this branch for leap and non-leap years?

My expectation is that this will turn out to be fragile in ways unrelated to leap vs non-leap.  My preference would be to keep an eye out for from-the-wild cases that hit these branches rather than try to concoct them for the sake of coverage stats.

Reply:
That said, I do have a couple of PRs coming up that are just adding tests, many of which will currently be `xfail`s.

Reply:
I don't think  1996 vs 2000 matters. Just gave me a start because I was like, "Wait, that's not supposed to be a leap year because it's a century, actually, wait, the century is divisible by 4..." Doesn't matter to a computer and the comment is necessary to make it clear why you chose 2000 anyway.

Regarding the tests, I think it might be a good idea to come up with some tests that will hit those branches. I can do it if you didn't have anything in mind.

Reply:
Go for it.  Anything you don't get covered I'll circle back to get next week.